src/validation/admin/bank-details-schema/bank-details-schema.ts

```diff
@@ -0,0 +1,127 @@
+import { DONATE_VALIDATION } from '../../../const/admin/donate';
+import * as Yup from 'yup';
+
+export interface BankDetailsValidationContext {
+    isPublishing: boolean;
+}
+
+export const BankDetailsValidationSchema = Yup.object({
+    name: Yup.string().trim().required(DONATE_VALIDATION.name.getRequiredError()),
+    receiver: Yup.string().trim().required(DONATE_VALIDATION.receiver.getRequiredError()),
+    edrpou: Yup.string()
+        .matches(/^\d+$/, DONATE_VALIDATION.getDigitsOnlyError())
+        .min(DONATE_VALIDATION.edrpou.count, DONATE_VALIDATION.edrpou.getMinError())
+        .max(DONATE_VALIDATION.edrpou.count, DONATE_VALIDATION.edrpou.getMaxError())
+        .required(DONATE_VALIDATION.edrpou.getRequiredError()),
+    iban: Yup.string()
+        .matches(/UA/, DONATE_VALIDATION.iban.getRequiredError())
+        .matches(/^UA\d+$/, DONATE_VALIDATION.getDigitsOnlyError())
+        .min(DONATE_VALIDATION.iban.count, DONATE_VALIDATION.iban.getMinError())
+        .max(DONATE_VALIDATION.iban.count, DONATE_VALIDATION.iban.getMaxError()),
+    paymentPurpose: Yup.string().trim().required(DONATE_VALIDATION.paymentPurpose.getRequiredError()),
+    swift: Yup.string()
+        .trim()
+        .required(DONATE_VALIDATION.swift.getRequiredError())
+        .min(DONATE_VALIDATION.swift.count, DONATE_VALIDATION.swift.getMinError())
+        .max(DONATE_VALIDATION.swift.count, DONATE_VALIDATION.swift.getMaxError()),
+    address: Yup.string().trim().required(DONATE_VALIDATION.address.getRequiredError()),
+    account: Yup.string().trim().required(DONATE_VALIDATION.account.getRequiredError()),
+});
+
+export const SupportOptionsValidationSchema = Yup.object({
+    name: Yup.string().trim().required(DONATE_VALIDATION.getDigitsOnlyError()),
+    value: Yup.string().trim().required(DONATE_VALIDATION.getDigitsOnlyError()),
+});
+
```

Comment:
Create `createValidator` in a separate file, because it will be needed for the other schema files
```suggestion
function createValidator<T extends Yup.AnyObjectSchema>(
    schema: T,
    field: keyof T['fields']
) {
    return (value: string): string | undefined => {
        try {
            schema.validateSyncAt(field as string, { [field]: value });
            return undefined;
        } catch (error: unknown) {
            if (error instanceof Yup.ValidationError) {
                return error.message;
            }
            console.error(`Unexpected validation error at ${field as string}:`, error); // Think if we need display it in console
            return 'An unexpected validation error occurred.';
        }
    };
}
```